compiler/src/dotty/tools/dotc/ast/Desugar.scala

```diff
@@ -1209,6 +1209,38 @@ object desugar {
       case _ => body
     cpy.PolyFunction(tree)(tree.targs, stripped(tree.body)).asInstanceOf[PolyFunction]
 
+  /** Desugar [T_1 : B_1, ..., T_N : B_N] => (P_1, ..., P_M) => R
+   *  Into    [T_1, ..., T_N] => (P_1, ..., P_M) => (B_1, ..., B_N) ?=> R
+   */
+  def expandPolyFunctionContextBounds(tree: PolyFunction)(using Context): PolyFunction =
```

Comment:
I was wondering whether we could re-use some of the code that desugars context bounds in methods. But maybe not. In any case, need to check that all applicable cases there are also present here.